scylla-gdb.py

```diff
@@ -2790,6 +2795,93 @@ def invoke(self, arg, for_tty):
 
 
+class scylla_io_queues(gdb.Command):
+    """ Print a summary of the reactor's IO queues.
+
+    Example:
+    Dev 0:
+        Class:                  |shares:         |ptr:            
+        --------------------------------------------------------
+        "default"               |1               |0x6000002c6500  
+        "commitlog"             |1000            |0x6000003ad940  
+        "memtable_flush"        |1000            |0x6000005cb300  
+        "streaming"             |200             |0x0             
+        "query"                 |1000            |0x600000718580  
+        "compaction"            |1000            |0x6000030ef0c0  
+
+        Max request size:    2147483647
+        Max capacity:        Ticket(weight: 4194303, size: 4194303)
+        Capacity tail:       Ticket(weight: 73168384, size: 100561888)
+        Capacity head:       Ticket(weight: 77360511, size: 104242143)
+
+        Resources executing: Ticket(weight: 2176, size: 514048)
+        Resources queued:    Ticket(weight: 384, size: 98304)
+        Handles: (1)
+            Class 0x6000005d7278:
+                Ticket(weight: 128, size: 32768)
+                Ticket(weight: 128, size: 32768)
+                Ticket(weight: 128, size: 32768)
+        Pending in sink: (0)
+
+
+    """
+    def __init__(self):
+        gdb.Command.__init__(self, 'scylla io-queues', gdb.COMMAND_USER, gdb.COMPLETE_NONE, True)
+
+    class ticket:
+        def __init__(self, ref):
+            self.ref = ref
+
+        def __str__(self):
+            return f"Ticket(weight: {self.ref['_weight']}, size: {self.ref['_size']})"
+
+    @staticmethod
+    def _print_io_priority_class(pclass_ptr, names_from_ptrs, indent = '\t\t'):
+        pclass = seastar_lw_shared_ptr(pclass_ptr).get().dereference()
+        gdb.write(f"{indent}Class {names_from_ptrs.get(pclass.address, pclass.address)}:\n")
+        slist = intrusive_slist(pclass['_queue'])
+        for entry in slist:
+            gdb.write(f"{indent}\t{scylla_io_queues.ticket(entry['_ticket'])}\n")
+
+    def invoke(self, arg, for_tty):
+        for dev, ioq in get_local_io_queues():
+            gdb.write(f"Dev {dev}:\n")
+
+            names = std_array(ioq['_registered_names'])
+            shares = std_array(ioq['_registered_shares'])
+            pclasses = std_vector(ioq['_priority_classes'])
+
+            names_from_ptrs = {}
+
+            gdb.write("\t{:24}|{:16}|{:16}\n".format("Class:", "shares:", "ptr:"))
+            gdb.write("\t" + '-'*56 + "\n")
+            for i, pclass in enumerate(pclasses):
+                pclass_ptr = std_unique_ptr(pclass).get()
+                names_from_ptrs[pclass_ptr] = names[i]
+                gdb.write("\t{:24}|{:16}|{:16}\n".format(str(names[i]), str(shares[i]), str(pclass_ptr)))
+            gdb.write("\n")
+
+            group = std_shared_ptr(ioq['_group']).get().dereference()
+            gdb.write(f"\tMax request size:    {group['_maximum_request_size']}\n")
```

Comment:
This breaks python2 compatibility, which is something we aim to maintain so long as we support python2 only distributions, like centos7. This of no concern if these io-queues are not present in versions that run on said OSs.

In any case this can be solved in a follow-up.

Reply:
Alright... the current source starts by declaring python3 as the interpreter, so I thought that Python 2 was officially deprecated for scylla-gdb:
https://github.com/scylladb/scylla/blob/f2ecb4617e479c9803e589ffd1b1e92abc47f472/scylla-gdb.py#L1

... but it's a simple-enough translation, so I guess there's no harm in maintaining compatibility.

Reply:
Yes, the main target is python3, as that is *the* python version we will eventually only support. I hope we can drop the python2 restriction soon.

Reply:
Please convert the below lines to `"".format()` as well, with that LGTM.